### PR TITLE
Feature/status field validation

### PR DESCRIPTION
**Pull request by gevans-ch**
Resolves: https://companieshouse.atlassian.net/browse/BI-10445 

Added validation for Status field on PscDiscrepancyReports filed by API filing customers
Added Unit tests for new validation

### CODE REVIEW

**Comment by mreis-ch** [comment]
> 

The setErrorMsgToLocation method in both the `PscDiscrepancyReportValidator` and `PscDiscrepancyValidator` should be removed probably as the same method exists in their parent `Validators`

File: src/test/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyReportValidatorUnitTest.java [hunk 1]
```diff
@@ -358,6 +381,53 @@ void unsuccessful_NullStatus() {
...
+                    Errors errorsFromValidation = pscDiscrepancyReportValidator.validateForUpdate(pscDiscrepancyReport,
+                        updatedReport);
+
+                    assertEquals(1, errorsFromValidation.size());
+                    assertTrue(errorsFromValidation.containsError(invalidStatus));
+                }
+
+                @Test
+                @DisplayName("Validate unsuccessful update of a PscDiscrepancyReport - validateStatusUpdate Already COMPLETE report")
+                void unsuccessful_AlreadyCompleteStatusReport() {
```
Comment (mreis-ch) [inline src/test/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyReportValidatorUnitTest.java:406]: Think this should be referencing it's an already SUBMITTED report maybe?

**Review by mreis-ch — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyReportValidator.java [hunk 2]
```diff
@@ -123,6 +127,29 @@ public Errors validateForUpdate(final PscDiscrepancyReport preexistingReport,
      * @param status String representation of the status.
      * @return Errors object containing any errors
      */
+    private Errors validateStatusCreation(final Errors errors, final String status) {
+        if(validateStatus(errors, status).hasErrors()){
+            return errors;
+        } else if(!status.equalsIgnoreCase(ReportStatus.INCOMPLETE.toString())){
```
[the remark this answers is not in the document]
Comment (gevans-ch) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyReportValidator.java:133]: Makes sense

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyReportValidator.java [hunk 3]
```diff
@@ -30,6 +30,10 @@
     private static final String SUBMISSION_REFERENCE = "submission_reference";
     private static final String INVALID_CHAR_MESSAGE = " contains an invalid character";
 
+    private static final String ALREADY_SUBMITTED_MESSAGE = "report has already been submitted";
```
Comment (mreis-ch) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/validation/PscDiscrepancyReportValidator.java:33]: The error message on the ticket is "**the report has already been submitted**". Think the _the_ should be _The_

**Review by mreis-ch — APPROVED** [review]

**Pull request merged**